tests/Stoa.test.ts

```diff
@@ -118,12 +124,140 @@ describe ('Test of Stoa API Server', () =>
             })
             .finally(doneIt);
     });
+
+    it ('Tests that sending a pre-image with get /validator and /validators', async () =>
+    {
+        let uri = URI(host)
+            .port(port)
+            .directory("preimage_received");
+        let response = await client.post (uri.toString(), {pre_image: sample_preImageInfo});
+        assert.strictEqual(response.status, 200);
+
+        let uri2 = URI(host)
+            .port(port)
+            .directory("validator")
+            .filename("GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2OKDM2VJ235GN")
+            .setSearch("height", "7");
+
+        response = await client.get (uri2.toString());
+        assert.strictEqual(response.data.length, 1);
+        assert.strictEqual(response.data[0].preimage.distance, 6);
+        assert.strictEqual(response.data[0].preimage.hash,
+            "0x4869b90d82af612dac15b6f152700b2e0f0b4a198fa09d83853d4ac3be4032b051c48806692b37776534f2ae7b404c9221ae1c9616fe50e3585d63e607d0afc6");
+
+        let uri3 = URI(host)
+            .port(port)
+            .directory("validator")
+            .filename("GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2OKDM2VJ235GN")
+            .setSearch("height", "1");
+        response = await client.get (uri3.toString());
+        assert.strictEqual(response.data.length, 1);
+        assert.strictEqual(response.data[0].preimage.distance, 0);
+        assert.strictEqual(response.data[0].preimage.hash,
+            "0xba665738077b352ed93c2d30882bd0505cf1147ed7610fd43d8bfe72cb29eee3b9b95a81bb3550c23dfa811bde4a7290d1dba85097b064de3557878fe62fd6ab");
+
+        let uri4 = URI(host)
+            .port(port)
+            .directory("validator")
+            .filename("GA3DMXTREDC4AIUTHRFIXCKWKF7BDIXRWM2KLV74OPK2OKDM2VJ235GN")
+            .setSearch("height", "8");
+        response = await client.get (uri4.toString());
+        assert.strictEqual(response.data.length, 1);
+        assert.strictEqual(response.data[0].preimage.distance, null);
+        assert.strictEqual(response.data[0].preimage.hash, Hash.NULL);
+
+        let uri5 = URI(host)
+            .port(port)
+            .directory("validators");
+        response = await client.get (uri5.toString());
+        assert.strictEqual(response.data.length, 3);
+        assert.strictEqual(response.data[0].preimage.distance, 0);
+        assert.strictEqual(response.data[0].preimage.hash,
+            "0xba665738077b352ed93c2d30882bd0505cf1147ed7610fd43d8bfe72cb29eee3b9b95a81bb3550c23dfa811bde4a7290d1dba85097b064de3557878fe62fd6ab");
+
+        let block = new Block();
+        let height = new Height();
+        let enrollment = new Enrollment();
+        height.value = 1008;
+        block.header.height = height;
+
+        // re-enrollment
+        enrollment.cycle_length = 1008;
+        enrollment.utxo_key =
+            "0x210b66053c73e7bd7b27673706f0272617d09b8cda76605e91ab66ad1cc3bfc1f3f5fede91fd74bb2d2073de587c6ee495cfb0d981f03a83651b48ce0e576a1a";
```

Comment:
Where did this validator previously enroll ? I can't find this hash anywhere.

Reply:
Here it is.
https://github.com/bpfkorea/stoa/blame/v0.x.x/tests/SampleData.test.ts#L29

Reply:
It's registered here.
https://github.com/bpfkorea/stoa/blob/51c52171fc60062235cb74588a6dd6c824c5c77e/tests/Stoa.test.ts#L79